src/aihwkit/utils/analog_info.py

```diff
@@ -0,0 +1,203 @@
+        """Compute the reuse factor.
+
+        The reuse factor is the number of vector matrix multiplication a layer computes."""
+        if isinstance(self.module, AnalogLinear):
+            ruf = reduce(operator.mul, (self.input_size), 1) // int(self.input_size[-1])
+            self.__set_reuse_factor(ruf)
+        elif isinstance(self.module, _AnalogConvNd):
+            # TODO: Extend the formula to ConvNd
+            ruf = reduce(operator.mul, (self.output_size), 1) // self.output_size[1]
+            self.__set_reuse_factor(ruf)
+
+    def layer_summary_dict(self) -> dict:
+        """Return a dictionary with all layer's information."""
+
+        analog = "analog" if self.isanalog else "digital"
+        return {"name": self.name,
+                "isanalog": analog,
+                "input_size": str(self.input_size) if self.input_size is not None else "-",
+                "output_size": str(self.output_size) if self.output_size is not None else "-",
+                "kernel_size": str(self.kernel_size) if self.kernel_size is not None else "-",
+                "num_tiles": self.num_tiles,
+                "macs": "",
+                "reuse_factor": str(self.reuse_factor) if self.reuse_factor is not None else "-"}
+
+    def __repr__(self) -> str:
+        """Print layer's information in the summary table."""
+        stats = self.layer_summary_dict().values()
+        result = ("{:<15} {:<15} {:<15} {:<15} {:<15} {:<20} {:<15} {:<15}\n".format(*stats))
+        return result
+
+
+class AnalogInfo:
+    """class for computing and storing results of the analog summary."""
+    def __init__(self,
+                 model: nn.Module,
+                 input_size: INPUT_SIZE_TYPE = None,
+                 rpu_config: Optional[RPUConfigAlias] = None):
+
+        self.model = model
+        self.input_size = input_size
+        self.rpu_config = rpu_config
+        self.layer_summary = self.create_layer_summary()
+        self.total_tile_number = self.calculate_num_tiles()
+        self.set_layer_sizes()
+
+    def set_layer_sizes(self) -> None:
+        """Set each layer input, output and kernel sizes."""
+        parent_input_size = self.input_size
+        for layer in self.layer_summary:
+            layer.calculate_size(parent_input_size)
+            parent_input_size = layer.output_size
+            layer.set_kernel_size()
+            layer.calculate_reuse_factor()
+
+    def create_layer_summary(self) -> List[LayerInfo]:
+        """Create the layer summary list.
+
+        This list contains LayerInfo elements that corresponds to each layer of the model."""
+        layer_summary = []
+        if isinstance(self.model, tuple):
+            layer_summary.append(LayerInfo(self.model[0], self.rpu_config))
+        else:
+            for module in self.model.modules():
+                if not isinstance(module, nn.Sequential):
+                    layer_summary.append(LayerInfo(module, self.rpu_config))
```

Comment:
Each tile has its own `rpu_config` in principle. So best would be if the rpu_config for a tile is taken here. Shortly, I will update the master branch with a new commit that can loop over analog_tiles() for an AnalogModuleBase. One could use that one.